internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go

```diff
@@ -0,0 +1,59 @@
+package cluster_mgrs
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/internal/kafka/internal/config"
+	"github.com/golang/glog"
+
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/workers"
+	"github.com/google/uuid"
+)
+
+const (
+	DynamicScaleUpWorkerType = "dynamic_scale_up"
+)
+
+type DynamicScaleUpManager struct {
+	workers.BaseWorker
+
+	DataplaneClusterConfig *config.DataplaneClusterConfig
+}
+
+var _ workers.Worker = &DynamicScaleUpManager{}
+
+func NewDynamicScaleUpManager(
+	reconciler workers.Reconciler,
+	dataplaneClusterConfig *config.DataplaneClusterConfig,
+) *DynamicScaleUpManager {
+
+	return &DynamicScaleUpManager{
+		BaseWorker: workers.BaseWorker{
+			Id:         uuid.New().String(),
+			WorkerType: DynamicScaleUpWorkerType,
+			Reconciler: reconciler,
+		},
+		DataplaneClusterConfig: dataplaneClusterConfig,
+	}
+}
+
+func (m *DynamicScaleUpManager) Start() {
+	m.StartWorker(m)
+}
+
+func (m *DynamicScaleUpManager) Stop() {
+	m.StopWorker(m)
+}
+
+func (m *DynamicScaleUpManager) Reconcile() []error {
+	if !m.DataplaneClusterConfig.IsDataPlaneAutoScalingEnabled() {
```

Comment:
For now we always run the reconciler and skip the reconciling depending on the autoscaling mode.

It would be nice to investigate a way to not create the reconciler based on that